syncer/syncer.py

```diff
@@ -193,19 +243,18 @@ def fill_bucket(policy_id, fill_interval_s, fill_quantity, capacity, scheduled_a
             scheduled_at,
         )
         scheduler.enter(fill_interval_s, PRIORITY, fill_bucket, argument=arguments)
+
+    if refresh_buckets_sec is not None:
+        # Schedule refreshing buckets with values from sentinel hub
+        arguments = (rate_limits, auth_token)
+        scheduler.enter(refresh_buckets_sec, PRIORITY_REFRESH_BUCKETS, refresh_buckets, argument=arguments)
+        logging.info(f"Refreshing buckets every {refresh_buckets_sec} seconds.")
+
     scheduler.run()
 
 
 def main():
-    CLIENT_ID = os.environ.get("CLIENT_ID")
-    CLIENT_SECRET = os.environ.get("CLIENT_SECRET")
-    # Docker-compose doesn't strip double quotes when reading from .env; however running this file from
-    # command line decodes the secret incorrectly if the quotes are absent. To avoid having two different
-    # ways of writing .env files, we remove the quotes here if present:
-    if CLIENT_SECRET.startswith('"') and CLIENT_SECRET.endswith('"'):
-        CLIENT_SECRET = CLIENT_SECRET[1:-1]
-    if not CLIENT_ID or not CLIENT_SECRET:
-        raise Exception("Please supply CLIENT_ID and CLIENT_SECRET env vars!")
+    REFRESH_BUCKETS_SEC = int(os.environ.get("REFRESH_BUCKETS_SEC", 500))
```

Comment:
This fails if `REFRESH_BUCKETS_SEC` env var is not set.

```bash
syncer    | Traceback (most recent call last):
syncer    |   File "syncer.py", line 281, in <module>
syncer    |     main()
syncer    |   File "syncer.py", line 257, in main
syncer    |     REFRESH_BUCKETS_SEC = int(os.environ.get("REFRESH_BUCKETS_SEC", 500))
syncer    | ValueError: invalid literal for int() with base 10: ''
syncer exited with code 1
```

Reply:
Thanks, I didn't realise docker-compose automatically sets it to a blank string if not available. Having a default was wrong anyway.

Reply:
Works now.